### PR TITLE
Update Go to 1.21.5 and k8s.io/* to 0.29.1

### PR DESCRIPTION
**Pull request by empovit**
This PR also incorporates #14 to fix broken APIs.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,82 +1,88 @@
 module github.com/NVIDIA/k8s-dra-driver
 
-go 1.20
+go 1.21.5
```
Comment (elezar) [inline go.mod:3]:
```suggestion
go 1.21
```

File: cmd/nvidia-dra-plugin/driver.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/util/retry"
 	"k8s.io/klog/v2"
-	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
```
Comment (klueska) [inline cmd/nvidia-dra-plugin/driver.go:29]: I didn't think we could make this change until we also made the `kind` changes. My expectation was that this PR would pull in the controller changes from #14 but not the kubelet-plugin changes. I'd also prefer to see this as a sparate commit, not bundled in the same commit as the go.mod changes.

**Comment by elezar** [comment]
@empovit I believe that this PR is no longer required since #77 and #14 have been merged.

**Comment by empovit** [comment]
Fixed by #77 and #14

**Pull request closed without merging**